cmd/juno/cli/create_dynamic_config.go

```diff
@@ -0,0 +1,188 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/NethermindEth/juno/internal/config"
+	"github.com/gdamore/tcell/v2"
+	"github.com/rivo/tview"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v2"
+)
+
+// updateConfig updates the configuration values from an instance of the Config struct.
+func updateConfig(newConfig config.Config) error {
+	file := viper.ConfigFileUsed()
+	data, err := yaml.Marshal(&newConfig)
+	if err != nil {
+		return err
+	}
+	err = os.WriteFile(file, data, 0o644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// createDynamicConfigCmd represents the createDynamicConfig command
+var createDynamicConfigCmd = &cobra.Command{
+	Use:   "configure",
+	Short: "Update your juno configuration file",
+	Long: `Available fields:
+		- Enable/Disable:
+			- RPC Server 
+			- Node Metrics Server
+			- Node REST API
+			- API Synchronization Mode
+
+		- Ethereum Node URL
+		- Network ID
+
+		Advanced:
+		- Port Settings
+		- Logger Verbosity Settings
+	.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		main()
+	},
+}
+
+func main() {
+	app := tview.NewApplication()
+	runtime := *config.Runtime
+	newConfig := &runtime
+
+	networkList := []string{"mainnet", "goerli"}
+	selectedNw := runtime.Starknet.Network
+	// Find index of selected network inside the networkList
+	selectedNwIndex := -1
+	for i, nw := range networkList {
+		if nw == selectedNw {
+			selectedNwIndex = i
+			break
+		}
+	}
+	verbosityList := []string{"DEBUG", "INFO", "WARN", "ERROR", "DPANIC", "PANIC", "FATAL"}
+	selectedVerb := runtime.Logger.VerbosityLevel
+	// Find index of selected verbosity level inside the verbosityList
+	selectedVerbIndex := -1
+	for i, verb := range verbosityList {
+		if verb == selectedVerb {
+			selectedVerbIndex = i
+			break
+		}
+	}
```

Comment:
I don't see the current log that I have set, I think you need to do all these checks with string lowercase, and that will fix the problem

Reply:
Solved with strings.EqualFold - thanks!